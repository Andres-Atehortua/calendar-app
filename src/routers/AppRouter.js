import { BrowserRouter as Router, Redirect, Switch } from 'react-router-dom';
import LoginRegisterScreen from '../components/auth/LoginRegisterScreen';
import { useDispatch, useSelector } from 'react-redux';
import CalendarScreen from '../components/calendar/CalendarScreen';
import PrivateRoutes from './PrivateRoutes';
import PublicRoutes from './PublicRoutes';
import { useEffect } from 'react';
import { startChecking } from '../redux/actions/authActions';

const AppRouter = () => {
  const dispatch = useDispatch();
  const { checking, uid } = useSelector((state) => state.auth);

  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);

  if (checking) return <h1>Cargando...</h1>;

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoutes isAuthenticated={!!uid} path='/login'>
            <LoginRegisterScreen />
          </PublicRoutes>

          <PrivateRoutes isAuthenticated={!!uid} exact path='/'>
            <CalendarScreen />
          </PrivateRoutes>

          <Redirect to='/' />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
