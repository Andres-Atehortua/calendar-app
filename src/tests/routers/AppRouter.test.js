import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import AppRouter from '../../routers/AppRouter';
const middlewares = [thunk];

const mockStore = configureStore(middlewares);

describe('Testing AppRouter', () => {
  test('should match snapshot', () => {
    const initStore = {
      auth: {
        checking: true,
      },
    };

    let store = mockStore(initStore);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').exists()).toBeTruthy();
  });

  test('should show public routes', () => {
    const initStore = {
      auth: {
        uid: null,
      },
    };

    let store = mockStore(initStore);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );

    expect(wrapper.find('.login-container').exists()).toBeTruthy();
  });

  test('should show private routes', () => {
    const initStore = {
      auth: {
        uid: '123-abc',
        name: 'ficticio',
      },
      calendar: {
        events: [],
      },
      ui: {
        openModal: false,
      },
    };

    let store = mockStore(initStore);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );

    expect(wrapper.find('.calendar__container').exists()).toBeTruthy();
  });
});
