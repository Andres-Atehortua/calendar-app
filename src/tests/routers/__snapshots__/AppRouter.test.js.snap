// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AppRouter should match snapshot 1`] = `
<h1>
  Cargando...
</h1>
`;
